src/v/cloud_storage/remote_segment.cc

```diff
@@ -236,7 +236,11 @@ ss::future<> remote_segment::do_hydrate() {
     auto callback = [this](
                       uint64_t size_bytes,
                       ss::input_stream<char> s) -> ss::future<uint64_t> {
-        offset_index tmpidx(get_base_rp_offset(), get_base_kafka_offset(), 0);
+        offset_index tmpidx(
+          get_base_rp_offset(),
+          get_base_kafka_offset(),
+          0,
+          remote_segment_sampling_step_bytes);
```

Comment:
what is the `remote_segment_sampling_step_bytes` constant here? it seems like we are hydrating from disk so the sampling step should come from the on-disk metadata?

Reply:
I think this code path is triggered when we download the segment for the first time and have to rebuild the index, so we can choose which step to use.

Reply:
that makes sense if true